cpp/src/parquet/page_index.h

```diff
@@ -126,4 +130,75 @@ class PARQUET_EXPORT OffsetIndex {
   virtual const std::vector<PageLocation>& page_locations() const = 0;
 };
 
+/// \brief Interface for reading the page index from a Parquet row group.
+class PARQUET_EXPORT RowGroupPageIndexReader {
+ public:
+  virtual ~RowGroupPageIndexReader() = default;
+
+  /// \brief Read column index of a column chunk.
+  ///
+  /// \param[in] i column id of the column chunk.
+  /// \returns error Result if either column id is invalid or column index does not exist.
+  virtual ::arrow::Result<std::shared_ptr<ColumnIndex>> GetColumnIndex(int32_t i) = 0;
```

Comment:
Sorry if I missed this in prior reviews, but core parquet code uses exceptions generally instead of result.  It probably pays to be consistent.

Reply:
Also, maybe nullptr or optional should be returned if the column index doesn't exist, since this doesn't seem like an error case unless there is a different way for checking which columns have indeces present before this code is run.

Reply:
Use exception instead of status for the return type. Now it returns nullptr if page index does not exist and throw if column ordinal or block ordinal is out of bound.